Remotion/Core/Collections.DataStore/ExpiringDataStore.cs

```diff
@@ -26,15 +26,17 @@ namespace Remotion.Collections.DataStore
   /// The <see cref="ExpiringDataStore{TKey,TValue,TExpirationInfo,TScanInfo}"/> stores values that can be expire.
   /// </summary>
   public class ExpiringDataStore<TKey, TValue, TExpirationInfo, TScanInfo> : IDataStore<TKey, TValue>
+      where TKey: notnull
+      where TValue: notnull

```

Comment:
why does TValue have to be notnull?

Reply:
because of the argument check on line 69

Reply:
```
    public void Add (TKey key, TValue value)
    {
      ArgumentUtility.CheckNotNull ("key", key);
      ArgumentUtility.CheckNotNull ("value", value);
```

Reply:
Ah yes, and taken further, because the ExpirationPolicy requires a non-null value to calculate the expiration.